dart/lib/src/hub.dart

```diff
@@ -74,7 +75,7 @@ class Hub {
         'captureEvent called with null parameter.',
       );
     } else {
-      final item = _stack.first;
+      final item = _peek();
       if (item != null) {
         try {
           sentryId = await item.client.captureEvent(event, scope: item.scope);
```

Comment:
@rxlabz I noticed that all the methods are sync now because of `await`

> await: You can use the await keyword to get the completed result of an asynchronous expression. The await keyword only works within an async function.

should we aim for sync calls or keep it async?

@bruno-garcia

Reply:
I'm not sure I understand. They are not sync, they return a Future

Reply:
@rxlabz `item.client.captureEvent` returns a future, but you call this method using `await`
if you use the `await` keyboard, means you the future to be completed, so its a sync call, not async.
or I am not understanding the `await` keyword in Dart, which is also possible

Reply:
`await` is just used a "syntactic sugar", it doesn't change the nature of the code. Maybe https://dart.dev/codelabs/async-await will be clearer than me.
But I noticed that this methods is asynchronous when it calls item.client.captureEvent and synchronous if the hub is not enabled or if the event is null, we probably should change the signature to 
```dart
FutureOr<SentryId> captureEvent(Event event) async {//...}
```

Reply:
not a big fan because of:

> Note: the FutureOr<T> type is interpreted as dynamic in non strong-mode.

but I got it I guess:
`await item.client.captureEvent` makes a sync call, but the signature of the method
`hub.captureEvent` is also `async` so it's fine.

let's keep as it is till we have a concrete implementation of the transport layer, so we'd know what looks better.